src/OrchardCore.Modules/OrchardCore.GitHub/Startup.cs

```diff
@@ -27,18 +26,25 @@ public sealed class GitHubLoginStartup : StartupBase
 {
     public override void ConfigureServices(IServiceCollection services)
     {
-        services.AddSingleton<IGitHubAuthenticationService, GitHubAuthenticationService>();
         services.AddSiteDisplayDriver<GitHubAuthenticationSettingsDisplayDriver>();
         services.AddNavigationProvider<AdminMenuGitHubLogin>();
-        services.AddRecipeExecutionStep<GitHubAuthenticationSettingsStep>();
-
-        services.AddTransient<IConfigureOptions<GitHubAuthenticationSettings>, GitHubAuthenticationSettingsConfiguration>();
 
         // Register the options initializers required by the GitHub Handler.
-        // Orchard-specific initializers:
-        services.AddTransient<IConfigureOptions<AuthenticationOptions>, GitHubOptionsConfiguration>();
-        services.AddTransient<IConfigureOptions<GitHubOptions>, GitHubOptionsConfiguration>();
+        services.AddTransient<IConfigureOptions<AuthenticationOptions>, AuthenticationOptionsConfiguration>();
+
+        services.AddTransient<IConfigureOptions<GitHubAuthenticationOptions>, GitHubAuthenticationOptionsConfiguration>();
```

Comment:
@kevinchalet  shouldn't this be?

```suggestion
        services.AddTransient<IConfigureNamedOptions<GitHubAuthenticationOptions>, GitHubAuthenticationOptionsConfiguration>();
```

Also, we we still need the `services.AddTransient<IPostConfigureOptions<GitHubAuthenticationOptions>, OAuthPostConfigureOptions<GitHubAuthenticationOptions, GitHubAuthenticationHandler>>();` line?

Reply:
> @kevinchalet shouldn't this be?

No, `OptionsFactory` resolves instances of type `IConfigureOptions<T>`, not `IConfigureNamedOptions<T>`: https://github.com/dotnet/runtime/blob/264531cbe53ed21b3ed89e05b23a25391bdd509b/src/libraries/Microsoft.Extensions.Options/src/OptionsFactory.cs#L36. Let's not change that 😄 

> Also, do we we still need the following line?

Yes, that post-initializer is provided by ASP.NET Core and is 100% required (otherwise, things like the backchannel HTTP handler or the `state` ASP.NET Core Data Protection formatter won't be attached).

Reply:
Then why do we need the `IConfigureNamedOptions<T>` implementation? shouldn't it just be `IConfigureOptions<T>`

Reply:
https://github.com/OrchardCMS/OrchardCore/issues/17631#issuecomment-2727642904

Reply:
@kevinchalet so you good with this PR as is? Can we :shipit: ?